conans/client/installer.py

```diff
@@ -455,6 +455,7 @@ def _propagate_info(node):
 
     def _call_package_info(self, conanfile, package_folder, ref):
         conanfile.cpp_info = CppInfo(package_folder)
+        conanfile.cpp_info.name = conanfile.name
```

Comment:
This name has to be assigned here before so the user can later change this name and the value is preserved. Setting this name in deps_cpp_info.update() will overwrite the name value when the info is propagated